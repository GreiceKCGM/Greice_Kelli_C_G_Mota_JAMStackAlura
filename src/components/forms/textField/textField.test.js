import React from 'react';
import user from '@testing-library/user-event';
import { render, screen } from '../../../infra/test/testUtils';
import TextField from './index';

describe('<TextField />', () => {
  test('renders component', () => {
    render(
      <TextField
        placeholder="Nome"
        value="Greice"
        onChange={() => {}}
        name="nome"
      />,

    );

    // screen.debug();

    const textField = screen.getByPlaceholderText(/nome/i);

    expect(textField).toMatchSnapshot();
  });
  describe('when field is valid', () => {
    describe('and user is typing', () => {
      test('the value must be updated', () => {
        const onChangeMock = jest.fn();
        const valueMock = 'greicekelly@gmail.com';
        render(
          <TextField
            placeholder="Nome"
            value=""
            onChange={onChangeMock}
            name="name"
            isTouched
          />
          ,
        );

        const nomeField = screen.getByPlaceholderText(/nome/i);
        user.type(nomeField, valueMock);
        expect(onChangeMock).toHaveBeenCalledTimes(valueMock.length);
      });
    });
  });

  describe('when field is invalid', () => {
    test('displays the respective error message', () => {
      render(
        <TextField
          placeholder="Email"
          value="greicekelly@gmail.com"
          onChange={() => {}}
          name="email"
          isTouched
          error="Digite um e-mail válido"
        />,
      );

      const fieldEmail = screen.getByPlaceholderText(/email/i);
      expect(fieldEmail).toHaveValue('greicekelly@gmail.com');
      expect(screen.getByRole('alert')).toHaveTextContent('Digite um e-mail válido');
      expect(fieldEmail).toMatchSnapshot();
    });
  });
});
