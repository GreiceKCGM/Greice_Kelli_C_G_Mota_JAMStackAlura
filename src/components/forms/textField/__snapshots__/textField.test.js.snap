// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextField /> renders component 1`] = `
.c0 {
  padding-bottom: inherit;
}

.c1 {
  width: 100%;
  border: 1px solid #b6b6b6;
  padding: 12px 16px;
  margin: 4px 0;
  padding-bottom: inherit;
  outline: 0;
  border-radius: 8px;
}

<input
  class="c0 c1"
  href=""
  name="nome"
  placeholder="Nome"
  type="text"
  value="Greice"
/>
`;

exports[`<TextField /> when field is invalid displays the respective error message 1`] = `
.c0 {
  padding-bottom: inherit;
}

.c1 {
  width: 100%;
  border: 1px solid #b6b6b6;
  padding: 12px 16px;
  margin: 4px 0;
  padding-bottom: inherit;
  outline: 0;
  border-radius: 8px;
  border-color: #dc3545;
}

.c1 + span {
  color: #dc3545;
  font-size: 11px;
}

<input
  class="c0 c1"
  href=""
  name="email"
  placeholder="Email"
  type="text"
  value="greicekelly@gmail.com"
/>
`;

exports[`<textField /> renders component 1`] = `
.c0 {
  padding-bottom: inherit;
}

.c1 {
  width: 100%;
  border: 1px solid #b6b6b6;
  padding: 12px 16px;
  margin: 4px 0;
  padding-bottom: inherit;
  outline: 0;
  border-radius: 8px;
}

<input
  class="c0 c1"
  href=""
  name="nome"
  placeholder="Nome"
  type="text"
  value="Greice"
/>
`;
